xla/backends/gpu/runtime/gpublas_lt_matmul_thunk.cc

```diff
@@ -135,10 +138,8 @@ CublasLtMatmulThunk::GetCachedMatmulPlan(const ExecuteParams& params) {
 
     TF_ASSIGN_OR_RETURN(auto plan,
                         blas_lt->GetMatmulPlan(gemm_config_, epilogue_));
-    // if workspace buffer is not provided, consider only the algorithms which
-    // do not require a scratch space
-    int64_t max_workspace =
-        workspace_.has_value() ? workspace_.value().size() : 0;
+    // Use the workspace size that was determined during autotuning
+    int64_t max_workspace = autotune_workspace_size_;
```

Comment:
Maybe I misunderstand something, but shouldn't we use the original (non-adjusted) workspace size when doing the query for GetAlgorithms down below? If we use the possibly smaller adjusted workspace size, some algorithms could be filtered out which can invalidate the algorithm index.

Reply:
Yes, the code is trying to use the same workspace size when thunk call GetAlgorithm as the autotuner.  But the comments is misleading, fixed the comments.